mewarpx/mewarpx/diags_store/particle_diagnostic.py

```diff
@@ -1,98 +1,104 @@
-"""Diagnostic code that wraps the picmi.ParticleDiagonstics class"""
+"""Diagnostic code that wraps the picmi.ParticleDiagnostics class"""
 
 import os
-import yt
 import glob
 import logging
+import warnings
 
 import matplotlib.pyplot as plt
-from mewarpx.diags_store.diag_base import WarpXDiagnostic
+import yt
 
 from pywarpx import callbacks, picmi
 
 from mewarpx.mwxrun import mwxrun
+from mewarpx.diags_store.diag_base import WarpXDiagnostic
 
 logger = logging.getLogger(__name__)
 
+
 class ParticleDiagnostic(WarpXDiagnostic):
     """Output particle diagnostics every diagnostic period and produce
-        plots from the diagnostics
+    plots from the diagnostics
 
     Contains:
         add_particle_diag (function): Function to add the particle diagnostic to
-        the mwxrun.simulation
+            the mwxrun.simulation
 
         post_processing: plot particle diagnostic data, not yet implemented
     """
 
     def __init__(self, diag_steps, name=None, species=None,
-                data_list=None, write_dir=".",
-                post_process=False, plot_data_list=None,
-                plot_species=None, **kwargs):
+                 data_list=None, write_dir='.', read_dir=None,
+                 post_process=False, plot_data_list=None,
+                 plot_species=None, **kwargs):
 
-        """
-        Arguments:
-            Initializes the picmi.ParticleDiagnostic and adds the diagnostic to
-            the simulation
+        """Initializes the picmi.ParticleDiagnostic and adds the diagnostic to
+        the simulation
 
+        Arguments:
             diag_steps (int): Number of steps between each diagnostic output
 
-            name (str): name of the diag output folder, defaults to diag
+            name (str): name of the diag output folder, defaults to
+                ``particle_diag``
 
-            species (mepicmi.Species): species in the simulation, if None then
-            uses all particles in the simulation
+            species (:class:`mewarpx.mepicmi.Species`): species in the
+                simulation, if None then uses all particles in the simulation
 
-            data_list (list str): list of attributes to be outputted by he
-            diagnostic, default uses ["position", "momentum", "weighting"]
+            data_list (list str): list of attributes to be outputted by the
+                diagnostic, default uses ``["position", "momentum", "weighting"]``
 
             write_dir (str): where to write particle diagnostic data
```

Comment:
Need documentation for read_dir, and explain how write_dir and read_dir are different.

Reply:
Is there ever any use case where they would be separate? `read_dir` is for where to read files written by `write_dir`...